cdc/changefeed.go

```diff
@@ -537,17 +548,19 @@ func (c *changeFeed) calcResolvedTs(ctx context.Context) error {
 
 	if minCheckpointTs > c.status.CheckpointTs {
 		c.status.CheckpointTs = minCheckpointTs
-		err := c.sink.EmitCheckpointTs(ctx, minCheckpointTs)
-		if err != nil {
-			return errors.Trace(err)
+		if c.ddlState != model.ChangeFeedWaitToExecDDL {
```

Comment:
why checking `c.ddlState != model.ChangeFeedWaitToExecDDL` here

Reply:
when `c.ddlState == model.ChangeFeedWaitToExecDDL`, the ddl may not executed yet, so we should wait and ensure the ddl is already executed.(here we just skip this checkpointTs)

Reply:
Oh, it means in this case `minCheckpointTs` can be updated to `c.ddlJobHistory[0].BinlogInfo.FinishedTS`, but DDL is waiting to be executed, so we can't forward `c.status.CheckpointTs`

Reply:
yes

Reply:
better to add some comments, or others will get confused

Reply:
done